File: CodeQuest2019 - Practice/Java/Prob18.java
import java.util.Scanner;

public class Prob18 {
    static final String key = "abcdefghijklmnopqrstuvwxyz";

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int ncases = Integer.valueOf(scanner.nextLine());
        for (int g = 0; g < ncases; g++) {
            String[] nums = scanner.nextLine().split(" ");
            float real = Float.valueOf(nums[0]);
            float imag = Float.valueOf(nums[1]);

            double zr = 0;
            double zi = 0;
            int n = 0;
            while (true) {
                zr = zr * zr - zi * zi + real;
                zi = 2 * zi * zr + imag;

                if (Math.pow(zr, 2) + Math.pow(zi, 2) > 100 * 100)
                    break;
                if (n > 50)
                    break;

                n++;
            }

            String color;
            //System.out.println(n);
            if (n <= 10)
                color = "RED";
            else if (n <= 20)
                color = "ORANGE";
            else if (n <= 30)
                color = "YELLOW";
            else if (n <= 40)
                color = "GREEN";
            else if (n <= 50)
                color = "BLUE";
            else
                color = "BLACK";

            String operator = "";
            if (zi >= 0)
                operator = "+";
            System.out.println(nums[0] + operator + nums[1] + "i " + color);
        }
    }
}

File: Elixir/Prob18.ex
defmodule Problem do
  def find_divergence(re, im, zr, zi, n) do
      fr = :math.pow(zr, 2) - :math.pow(zi, 2) + re
      fi = 2 * zi * zr + im

      cond do
        n > 50 -> n
        (:math.pow(fr, 2) + :math.pow(fi, 2) > :math.pow(100, 2)) -> n
        true -> find_divergence(re, im, fr, fi, n+1)
      end
  end

  def find_divergence(re, im) do
    find_divergence(re, im, 0, 0, 0)
  end

  def main() do
    [rst, ist] = IO.gets("")
    |> String.trim
    |> String.split(" ")

    {real, _} = Float.parse(rst)
    {imag, _} = Float.parse(ist)

    dn = find_divergence(real, imag)
    color = cond do
      dn <= 10 -> "i RED"
      dn <= 20 -> "i ORANGE"
      dn <= 30 -> "i YELLOW"
      dn <= 40 -> "i GREEN"
      dn <= 50 -> "i BLUE"
      true -> "i BLACK"
    end

    sign = if (imag >= 0) do
      "+"
    else
      ""
    end

    IO.puts(rst <> sign <> ist <> color)

  end
end


{n, "\n"} = IO.gets("")
  |> Integer.parse

for _ <- 1..n, do: Problem.main()
